movie_library/views/actor_controller.py

```diff
@@ -1,27 +1,74 @@
+from django.shortcuts import get_object_or_404
+from rest_framework import status
 from rest_framework.viewsets import ViewSet
 from rest_framework.response import Response
+from ..utils.custome_premissions import CustomPermissions
+from ..models.movie import Movie
 from ..models.actor import Actor
 from ..utils.serializers.actor_serializer import ActorSerializer
+from ..utils.serializers.actor_movies_serializer import ActorMoviesSerializer
+from ..models.movie_actor import MovieActor
 
 
 class ActorViewSet(ViewSet):
+    permission_classes = [CustomPermissions]
 
     def list(self, request):
         actors = Actor.objects.all()
         result = ActorSerializer(actors, many=True)
         return Response(data=result.data, status=200)
 
     def create(self, request):
-        pass
+        data = request.data
+        movies = data.get('movies')
+        actor = Actor(first_name=data.get('first_name'), last_name=data.get('last_name'), gender=data.get('gender'))
+        actor.save()
+        bulk_list = list()
+        query = Movie.objects.all()
+        for movie in movies:
+            the_movie = get_object_or_404(query, pk=movie)
+            #the_movie = Movie.objects.filter(pk=movie)
+            bulk_list.append(
+                MovieActor(actors=actor, movies=the_movie))
+
+        bulk_msj = MovieActor.objects.bulk_create(bulk_list)
+        result = ActorMoviesSerializer(actor, many=False)
+        return Response(data=result.data, status=201)
 
     def retrieve(self, request, pk=None):
-        pass
+        actors = Actor.objects.all()
+        actor = get_object_or_404(actors, pk=pk)
+        serializer = ActorMoviesSerializer(actor)
+        return Response(serializer.data, status=status.HTTP_200_OK)
 
     def update(self, request, pk=None):
-        pass
+        actors = Actor.objects.all()
+        actor = get_object_or_404(actors, pk=pk)
+        data = request.data
+        movies = data.get('movies')
+        actor.first_name = data.get('first_name')
+        actor.last_name = data.get('last_name')
+        actor.gender = data.get('gender')
+        actor.save()
```

Comment:
i am not sure if we are using fully update or partial update "we send all fields or just dirty" 
but better to move this to component/repo